resource/graph_node.go

```diff
@@ -361,6 +364,13 @@ func (w *GraphNode) SetNeedsUpdate() {
 	w.setNeedsReconfigure(w.Config(), false, w.UnresolvedDependencies())
 }
 
+// SetDisconnected is used to mark a remote node as disconnected.
+func (w *GraphNode) SetDisconnected() {
```

Comment:
IIUC, in the resource graph we have:
* A node for each remote robot. Where the underlying resource is the robot client
* A node for each resource on a robot robot. Where the underlying resource is the component client (e.g: camera client).

Does this `SetDisconnected` state only apply to nodes representing the remote robot client in the resource graph? Or does it also apply to the remote resource objects in the resource graph? 

Second:
And when a remote goes from working -> disconnected -> working, what state transition will that take? Is it ready -> disconnected -> ready? Or do we jump through any of the unconfigured/configuring states?

I think the answers would work well in the documentation.

Reply:
> Does this SetDisconnected state only apply to nodes representing the remote robot client in the resource graph? Or does it also apply to the remote resource objects in the resource graph?

it applies to the remote resource objects. also note that [resources representing full remote machine clients are excluded from the MachineStatus endpoint](https://github.com/viamrobotics/rdk/pull/4399/commits/2dadf77a5d5e061016b647d9fef6389ab50ae11b).

> And when a remote goes from working -> disconnected -> working, what state transition will that take? Is it ready -> disconnected -> ready? Or do we jump through any of the unconfigured/configuring states?

thanks for drawing attention to this. the flow I want is for disconnected nodes to go directly back into the ready state (unless they are unhealthy for some other reason). however, I don't think the current logic actually transitions disconnected nodes back to ready, so let me address that.

> I think the answers would work well in the documentation.

will do